mongo/integration/collection_test.go

```diff
@@ -1778,29 +1800,11 @@ func TestCollection(t *testing.T) {
 			assert.Equal(mt, int64(2), res.UpsertedCount, "expected %v upserted documents, got %v", 2, res.UpsertedCount)
 			assert.Equal(mt, 2, len(res.UpsertedIDs), "expected %v upserted ids, got %v", 2, len(res.UpsertedIDs))
 
-			// find the upserted documents and check their contents
-			id1, ok := res.UpsertedIDs[0]
+			// Check that IDs exist in result for upserted documents.
+			_, ok := res.UpsertedIDs[0]
 			assert.True(mt, ok, "expected id at key 0")
-			id2, ok := res.UpsertedIDs[int64(numModels-1)]
+			_, ok = res.UpsertedIDs[int64(numModels-1)]
 			assert.True(mt, ok, "expected id at key %v", numModels-1)
-
-			doc, err := mt.Coll.FindOne(mtest.Background, bson.D{{"_id", id1}}).DecodeBytes()
```

Comment:
I've removed the calls to `FindOne` to verify that the upserted documents exist. I _could_ mock responses again here, but that doesn't seem to be testing the functionality of a bulk write update with batches.

Reply:
I agree the `FindOne` does not seem necessary. This test was motivated by a bug in merging bulk update results: GODRIVER-1715